### PR TITLE
feat: add view3d-optimizer

### PR DESCRIPTION
**Pull request by WoodNeck**
## Details
- This adds `view3d-optimizer` 
- This fixes `.bin` export in view3d demo page

### CODE REVIEW

**Review by qpitlove — APPROVED** [review]

File: packages/view3d-optimizer/index.js [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+      const newTextureIndex = gltf.textures.length;
+
+      gltf.textures.push({
+        sampler: texture.sampler,
+        source: newTextureIndex
+      });
+
+      if (!texture.extras) texture.extras = {};
+      if (!texture.extras["view3d-lod"]) texture.extras["view3d-lod"] = [];
```
Comment (daybrush) [inline packages/view3d-optimizer/index.js:222]: I saw the code that uses the extra in view3d, it looks like an object containing an array of `levels`.

**Review by daybrush — APPROVED** [review]
LGTM

File: packages/view3d-optimizer/index.js (hunk 1 above)
Comment (WoodNeck) [reply packages/view3d-optimizer/index.js:222]: Oh well you're right, it should be an empty object 👍

**Review by malangfox — APPROVED** [review]
LGTM

**Pull request merged**